aiida/cmdline/commands/cmd_export.py

```diff
@@ -21,6 +22,7 @@
 from aiida.cmdline.utils import decorators
 from aiida.cmdline.utils import echo
 from aiida.common.links import GraphTraversalRules
+from aiida.tools.importexport import EXPORT_VERSION
```

Comment:
I think there was a reason behind moving all `importexport` module imports into the functions. I believe it had to do with  verdi's loading time? But if this works (all tests pass) then all good for me :+1:

Reply:
Ok, I see. Unfortunately, keeping the methods in that is not possible if we want to use the `EXPORT_VERSION` as the default value of one of the options.

However, there is an issue in the tests because it seems that importing `aiida.tools.importexport` mandates that `seekpath` is installed (which I consider a bug).
We'll need to find a solution for this.

Reply:
Fixing this in https://github.com/aiidateam/aiida-core/pull/4324

Reply:
> Ok, I see. Unfortunately, keeping the imports inside the methods is not possible if we want to use the `EXPORT_VERSION` as the default value of one of the click options.

Agreed, but since it's also not a necessity, it can be argued that it can be reverted again to simply say "latest version"?

> However, there is an issue in the tests because it seems that importing `aiida.tools.importexport` mandates that `seekpath` is installed (which I consider a bug).
> We'll need to find a solution for this.

Yes. I am unsure where this comes from exactly. What series of imports in the `importexport` module leads to the import of `seekpath`?

Reply:
Ok, after #4324 is merged, this now indeed triggers the load-time checker
```
Run verdi devel check-load-time
Critical: potential `verdi` speed problem: `aiida.orm` module is imported.
##[error]Process completed with exit code 1.
```
There are of course different ways of fixing this
 1. make sure that importing aiida.tools does not lead to importing aiida.orm 
 1. move the EXPORT_VERSION to a place from where it can safely be imported
 1. revert back the change to pass the EXPORT_VERSION as a default value of the option (change requested by @sphuber )

Given that tools are operating on AiiDA ORM classes and the docstring of the subpackage states that modules in this sub package may require the database environment to be loaded, I'm not sure option 1. makes much sense.

I would tend towards option 2., moving the EXPORT_VERSION to some high-level place (perhaps `aiida.common`?).
Let me know @sphuber @CasperWA

Reply:
I agree mostly with your sentiment here, although I would strongly advice against moving `EXPORT_VERSION` outside the `importexport` module.
However, we could create another file that simply stores this (and maybe other) hard-coded variables, which can then be imported directly from said file (i.e. one cannot do `from aiida.tools.importexport import EXPORT_VERSION` without also importing `aiida.orm`, but one can maybe do ` from aiida.tools.importexport.common.config import EXPORT_VERSION` or something?)

Reply:
> i.e. one cannot do from aiida.tools.importexport import EXPORT_VERSION without also importing aiida.orm, but one can maybe do from aiida.tools.importexport.common.config import EXPORT_VERSION or something?

Sure, that is what I am suggesting.

Reply:
Ah, I understood that you were suggesting `aiida.common.config`, i.e., not under the `importexport` module still, but rather the top-level `common`.

Reply:
So if I look closer, I fint that `EXPORT_VERSION` already is at `aiida.tools.importexport.common.config`. The mappings that depend on `orm` objects could then be moved from this to another file, maybe `utils` instead? That should then solve the problem (if you import `EXPORT_VERSION` from `aiida.tools.importexport.common.config`).

Reply:
Ah, sorry, I misread your comment (was a bit late ;-) ).

I think there is no way to prevent the import of `aiida.tools` when importing a submodule of it, i.e. the version constant would need to be moved out.

One thing I would note is that this issue is caused by the fact that we import the functionality from all tools directly into the name space `aiida.tools`:
https://github.com/aiidateam/aiida-core/blob/5f50f0335761386ddacbb5de4a6934b659b20e2d/aiida/tools/__init__.py#L24-L27

I get that it makes the import path for tools shorter, but it will also lead to nameclashes if one tool wants to provide a function with the same name as another tool (i.e. it forces us to encode the namespacing into function names).
Perhaps something worth reconsidering.

Reply:
What you're hinting at is done all over the code base, in all modules. It may be good to start moving away from wildcard imports, but I don't think I will be the one to make that change all throughout...

Reply:
In many places, I have done this intentionally though and I think is not necessarily bad practice. In the main modules (especially during the ORM rewrite of last year) I have used the approach of adding public facing interface in each module to the `__all__`. This then allows in the `__init__.py` of the corresponding module to use wildcard imports to import those and only those resources in the `__all__`. By doing this recursively all the way up to the second level module, e.g. `aiida.orm` all the public facing interface will be directly importable from there.

Reply:
Yes, I agree that it is not bad per se (e.g. when importing class names of node types that are meant to be unique), I just wonder whether we should be doing it in `aiida.tools`. 

Anyhow, to get back to the original problem: we need to decide - do we move out the EXPORT_VERSION constant or do we stop importing it?

Reply:
There is another alternative where you make a "lazy-default" which is just a callback that will only be called when the actual command is called. Only downside may be that I am not sure if this makes it possible to have the value show up in the help string. If it does, then that would also require loading the ORM and so incur the slow down, although that is then isolated to just that command which might be ok.

Reply:
I'd be fine with that.

Reply:
@sphuber Ah hm... I tried but it doesn't quite do what we had in mind.

When providing a callable to `default`, it ends up being shown as
```
  -v, --version VERSION           Archive format version to migrate to.
                                  [default: (dynamic)]
```
One can use `show_default='latest version'` to get 
```
  -v, --version VERSION           Archive format version to migrate to.
                                  [default: (latest version)]
```
but then we're almost back to square one anyhow.

I also tried the `contextual_default` but there the default value isn't shown in the `-h` output at all.

Given this, I think the less convoluted approach is not to set a default and to go back to the original logic.
Do you agree?

Reply:
For me this is fine (going back). I think I tried going through all this originally as well and thought I didn't want to bother untangling it for a feature that most people will not care about anyway. The main message is simply: "It will be the latest archive version" ... If one then wants to check that an archive indeed _is_ that latest version one could use the inspection functionality in verdi.

In the end, I agree that `aiida.tools` should maybe not include imports from submodules, but rather one should specify the individual tools. That makes more sense conceptually, I think.

Reply:
Agreed that the more convoluted method is not worth the trouble if it still doesn't get the fully desired result. So fine to revert. Please make a note in the commit explaining why this is and what we tried and why that didn't work. To hopefully prevent us from going on this wild goose chase once more in the future :)

Reply:
Done!